### PR TITLE
Update subprojects/wlroots

### PR DESCRIPTION
**Pull request by jbeich**
For wlroots debugging by comparing with `sway-git`. Briefly tested [downstream](https://github.com/freebsd/freebsd-ports/blob/2117c1cce9fc/x11-wm/hyprland/files/patch-wlroots-0.17).

Based on:
- https://github.com/swaywm/sway/pull/7637
- https://github.com/swaywm/sway/pull/7675

### CODE REVIEW

**Comment by jbeich** [comment]
Let's see if I can unbreak Nix without Nix to generate `narHash`.

**Comment by jbeich** [comment]
All green. `Run workflow` in `Nix update` did the trick after rebasing NVIDIA workaround.

**Review by vaxerski — APPROVED** [review]
k. I usually don't want wlroots bump mrs tho

**Comment by vaxerski** [comment]
planned on doing it anyways so w/e

File: src/render/Renderer.cpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ void renderSurface(struct wlr_surface* surface, int x, int y, void* data) {
 
     if (!g_pHyprRenderer->m_bBlockSurfaceFeedback) {
         wlr_surface_send_frame_done(surface, RDATA->when);
-        wlr_presentation_surface_sampled_on_output(g_pCompositor->m_sWLRPresentation, surface, RDATA->pMonitor->output);
+        wlr_presentation_surface_scanned_out_on_output(g_pCompositor->m_sWLRPresentation, surface, RDATA->pMonitor->output);
```
Comment (jbeich) [inline src/render/Renderer.cpp:60]: Oops, should have been:
```suggestion
        wlr_presentation_surface_textured_on_output(g_pCompositor->m_sWLRPresentation, surface, RDATA->pMonitor->output);
```

Comment (vaxerski) [reply src/render/Renderer.cpp:60]: fixed

File: nix/wlroots.nix [hunk 2]
```diff
@@ -44,10 +44,7 @@ assert (lib.assertMsg (hidpiXWayland -> enableXWayland) ''
           })
         ])
         ++ (lib.optionals nvidiaPatches [
-          (fetchpatch {
-            url = "https://aur.archlinux.org/cgit/aur.git/plain/0001-nvidia-format-workaround.patch?h=hyprland-nvidia-screenshare-git";
-            sha256 = "A9f1p5EW++mGCaNq8w7ZJfeWmvTfUm4iO+1KDcnqYX8=";
-          })
+          ./wlroots-nvidia.patch
```
Comment (fufexan) [inline nix/wlroots.nix:47]: May I ask what the reason for this change is?

Comment (jbeich) [reply nix/wlroots.nix:47]: AUR patch was:
- out-of-date, so didn't apply after [wlroots@c4d4409deb50](https://gitlab.freedesktop.org/wlroots/wlroots/-/commit/c4d4409deb50) back then (when tested by CI)
- referenced HEAD commit, so may not apply when using older Hyprland (when bisecting)

Comment (fufexan) [reply nix/wlroots.nix:47]: That makes sense, thanks for the explanation.

**Pull request merged**